src/DataAccess/DbalSubscriptionRepository.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+
+namespace WMDE\Fundraising\Frontend\DataAccess;
+
+use Doctrine\ORM\EntityManager;
+use WMDE\Fundraising\Entities\Subscription;
+use WMDE\Fundraising\Frontend\Domain\SubscriptionRepository;
+
+/**
+ * @license GNU GPL v2+
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
+ */
+class DbalSubscriptionRepository implements SubscriptionRepository {
+
+	private $entityManager;
+
+	public function __construct( EntityManager $entityManager ) {
+		$this->entityManager = $entityManager;
+	}
+
+	public function storeSubscription( Subscription $subscription ) {
+		$this->entityManager->persist( $subscription );
+		$this->entityManager->persist( $subscription->getAddress() );
+		$this->entityManager->flush();
+	}
+
+	public function countSimilar( Subscription $subscription, \DateInterval $interval ): int {
+		$qb = $this->entityManager->createQueryBuilder();
+		$now = new \DateTime();
```

Comment:
I know what you mean, but so far had no problems in my tests (Just create DateTime objects with relative times).
But if you want i can change the interface to datetime and put the cutoff calculation into the FunFunFactory.